### PR TITLE
Automatic localization integration

### PR DESCRIPTION
**Pull request by Astrono2**
I'm not quite sure this is optimal, I used PoolStringArray to optimize for memory, considering that ideally we'd have a lot of languages. But then there's the converting to Array to do anything. But it's only done once on startup, so idk.

I also hardcoded "en" once as default value when getting the saved locale and "English" thrice in the new `_set_languages` method. And I decided to put English at the top and sort the rest alphabetically, to help people find their language when eventually there's a lot of them.

There's also something that might or might not be a problem for future-proofing this for a big amount of languages:
image
Should I make it so that the list only grows downwards? Or is that okay?

### CODE REVIEW

File: lorien/Misc/I18nParser.gd [hunk 1]
```diff
@@ -25,7 +32,11 @@ func load_files() -> void:
 				else:
 					printerr("Key not found (make sure to use spaces; not tabs): %s" % line)
 			TranslationServer.add_translation(translation)
-		
+			locales.append(translation.locale)
+			language_names.append(name)
+	Settings.locales = locales
+	Settings.language_names = language_names
```
Comment (mbrlabs) [inline lorien/Misc/I18nParser.gd:38]: Don't set them here directly in the singleton. It's better to make an inner class like this in `I18nParser`:
````
class ParseResult:
     var locales: PoolStringArray
     var language_names: PoolStringArray
````

and return that.

File: lorien/Misc/I18nParser.gd [hunk 2]
```diff
@@ -5,12 +5,19 @@ const I18N_FOLDER := "res://Assets/I18n/"
...
+	var locales := PoolStringArray()
+	var language_names := PoolStringArray()
 	for f in _get_i18n_files():
 		var file := File.new()
 		print("Loading i18n file: %s" % f)
 		if file.open(f, File.READ) == OK:
 			var translation := Translation.new()
 			translation.locale = f.get_file().get_basename()
+			var name := file.get_line().strip_edges()
+			if not name.begins_with("LANGUAGE_NAME"):
```
Comment (mbrlabs) [inline lorien/Misc/I18nParser.gd:17]: Instead of `not` please use `!` here

File: lorien/UI/Dialogs/SettingsDialog.gd [hunk 3]
```diff
@@ -52,17 +47,34 @@ func _set_values() -> void:
...
 	_brush_size.value = brush_size
 	_brush_color.color = brush_color
 	_canvas_color.color = canvas_color
 	_project_dir.text = project_dir
 
+# -------------------------------------------------------------------------------------------------
+func _set_languages(current_locale: String) -> void:
+	# Technically, Settings.language_names is useless from here on out, but I figure it's probably gonna come in handy in the future
+	var sorted_languages := Array(Settings.language_names)
+	var unsorted_languages := sorted_languages.duplicate()
```
Comment (mbrlabs) [inline lorien/UI/Dialogs/SettingsDialog.gd:62]: What's the purpose of this unsorted array here? Can't you just use the sorted one?

**Review by mbrlabs — CHANGES_REQUESTED** [review]
Works great, thank you! I just left some comments, but looks good so far.

File: lorien/UI/Dialogs/SettingsDialog.gd (hunk 3 above)
Comment (Astrono2) [reply lorien/UI/Dialogs/SettingsDialog.gd:62]: The unsorted one is there to get the original index of a language before sorting alphabetically, by using `unsorted_languages.find()`.

**Comment by mbrlabs** [comment]
Alright, thank you for the changes. However, the language selection is now empty (except English):

Screenshot (2)

**Comment by mbrlabs** [comment]
@Astrono2 any chance you could fix the issue with the languages not showing up? I'd like to merge this PR before #25. If you don't have time, don't worry...i can merge it as is and fix it later myself 🙂

**Comment by Astrono2** [comment]
Oh sorry I didn't see your first reply, I'll fix it during the weekend.

**Review by mbrlabs — APPROVED** [review]

**Comment by mbrlabs** [comment]
Thanks!

**Pull request merged**